pkg/api/persistentvolumeclaim/util.go

```diff
@@ -39,26 +65,21 @@ func dataSourceInUse(oldPVCSpec *core.PersistentVolumeClaimSpec) bool {
 	if oldPVCSpec == nil {
 		return false
 	}
-	if oldPVCSpec.DataSource != nil {
+	if oldPVCSpec.DataSource != nil || oldPVCSpec.DataSourceRef != nil {
 		return true
 	}
 	return false
 }
 
 func dataSourceIsEnabled(pvcSpec *core.PersistentVolumeClaimSpec) bool {
```

Comment:
IsEnabled is now a total misnomer.

Reply:
DropDisabled is called by Create and Update, and Normalize is called by Create and Get/List. I'm not opposed to more refactoring, but I think we'll end up with a 3rd function and the unit tests will need a complete rewrite as a result.

I also agree that the existing method names suck, but there's a tension between wanting to clean up bad code and wanting to avoid fixing stuff that's not broken.

Reply:
It *is* broken, it's just broken in human-space not compiler-space :)